modules/nf-core/sourmash/compare/main.nf

```diff
@@ -0,0 +1,39 @@
+process SOURMASH_COMPARE {
+    label 'process_low'
+
+    conda (params.enable_conda ? "bioconda::sourmash=4.5.0" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/sourmash:4.5.0--hdfd78af_0':
+        'quay.io/biocontainers/sourmash:4.5.0--hdfd78af_0' }"
+
+    input:
+    path signatures
+    val save_numpy_matrix
+    val save_csv
+
+    output:
+    path "*comp"           , optional:true, emit: numpy_matrix
+    path "*comp.labels.txt", optional:true, emit: labels
+    path "*comp.csv"       , optional:true, emit: csv
+    path "versions.yml"    , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args   = task.ext.args     ?: ''
+    def comp   = save_numpy_matrix ? "--output comp"  : ''
+    def csv    = save_csv          ? "--csv comp.csv" : ''
```

Comment:
I would have also add a check making sure thatn one of either `csv` or `comp` are set to `true` (or add `[stdout](https://www.nextflow.io/docs/latest/process.html#output-type-stdout) emit: stdout`), otherwise a pipeline dev who quickly installs the module without looking closely will be confused why there is no output...